pages/blog/[slug].tsx

```diff
@@ -42,6 +43,25 @@ export default function BlogTemplate({
     editMode
   )
 
+  const cms = useCMS()
+
+  const saveToStorage = useCallback(formData => {
+    cms.api.storage.save(markdownFile.fileRelativePath, formData.values)
+  }, [])
+
+  // save to storage on change
+  useWatchFormValues(form, saveToStorage)
+
+  // load from storage on boot
+  useEffect(() => {
+    if (!editMode) return
+
+    const values = cms.api.storage.load(markdownFile.fileRelativePath)
+    if (values) {
+      form.updateValues(values)
+    }
+  }, [editMode])
```

Comment:
I suspect that if you navigate between blog the local-storage would fail. You might want to check that. Adding `form` to the deps list would fix that issue
```suggestion
  }, [form, editMode])
```

Reply:
I think you're right, it just so happens there's no direct navigation between blog posts on the site